src/AspNet.Security.OpenId/OpenIdAuthenticationHandler.cs

```diff
@@ -10,39 +10,55 @@
 using System.Linq;
 using System.Net.Http;
 using System.Security.Claims;
+using System.Text.Encodings.Web;
 using System.Threading.Tasks;
+using System.Web;
 using JetBrains.Annotations;
 using Microsoft.AspNetCore.Authentication;
-using Microsoft.AspNetCore.Http.Authentication;
 using Microsoft.AspNetCore.Http.Features.Authentication;
 using Microsoft.AspNetCore.WebUtilities;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace AspNet.Security.OpenId
 {
-    public class OpenIdAuthenticationHandler<TOptions> : RemoteAuthenticationHandler<TOptions> where TOptions : OpenIdAuthenticationOptions
+    public class OpenIdAuthenticationHandler<TOptions> : RemoteAuthenticationHandler<TOptions> where TOptions : OpenIdAuthenticationOptions, new()
     {
-        protected override async Task<AuthenticateResult> HandleRemoteAuthenticateAsync()
+        /// <summary>
+        /// The handler calls methods on the events which give the application control at certain points where processing is occurring. 
+        /// If it is not provided a default instance is supplied which does nothing when the methods are called.
+        /// </summary>
+        protected new OpenIdAuthenticationEvents Events
```

Comment:
As I said, move at the end, remove the XML documentation, make it get-only and private:

```
private new OpenIdAuthenticationEvents Events => (OpenIdAuthenticationEvents) base.Events;
```